e2e/testsuite/tx.go

```diff
@@ -137,10 +137,13 @@ If this is a compatibility test, ensure that the fields are being sanitized in t
 
 // ExecuteGovV1Proposal submits a v1 governance proposal using the provided user and message and uses all validators
 // to vote yes on the proposal. It ensures the proposal successfully passes.
-func (s *E2ETestSuite) ExecuteGovV1Proposal(ctx context.Context, msg sdk.Msg, chain *cosmos.CosmosChain, user ibc.Wallet, proposalID uint64) {
+func (s *E2ETestSuite) ExecuteGovV1Proposal(ctx context.Context, msg sdk.Msg, chain *cosmos.CosmosChain, user ibc.Wallet) {
 	sender, err := sdk.AccAddressFromBech32(user.FormattedAddress())
 	s.Require().NoError(err)
 
+	proposalID := s.proposalIds[chain.Config().ChainID]
+	s.proposalIds[chain.Config().ChainID]++
```

Comment:
good point, yes makes sense to do increment afterwards. In practice I think it will make no difference as the test will fail, but this would be a nice improvement!

Reply:
defer func() ? 

I reckon its probably fine either as you say, test will likely be failing